bokehjs/src/lib/models/annotations/box_annotation.ts

```diff
@@ -112,9 +114,10 @@ export class BoxAnnotationView extends AnnotationView {
     ctx.beginPath()
     ctx.rect(sleft, stop, sright-sleft, sbottom-stop)
 
-    this.visuals.fill.set_value(ctx)
-    ctx.fill()
-
+    if (this.visuals.fill.doit) {
+      this.visuals.fill.set_value(ctx)
+      ctx.fill()
+    }
     this.visuals.line.set_value(ctx)
```

Comment:
BTW if you want to make this really complete, you could add the same treatment to `visuals.line` and the `ctx.stroke` since it also should not render if the line color is  None